File: lollys-lotto-rust-sdk/src/instructions/cranks/crank_lotto_game_closed.rs
use lollys_lotto::instruction::CrankLottoGameClosed;

use crate::instructions::*;

pub fn crank_lotto_game_closed(
    round: u64,
    authority: &Pubkey,
    lotto_game: &Pubkey,
    event_emitter: &Pubkey,
) -> Instruction {
    let data = CrankLottoGameClosed { round }.data();
    let accounts = lollys_lotto::accounts::CrankLottoGameClosed {
        authority: *authority,
        lotto_game: *lotto_game,
        event_emitter: *event_emitter,
    }
    .to_account_metas(None);
    Instruction {
        program_id: lollys_lotto::ID,
        accounts,
        data,
    }
}

File: lollys-lotto-ts-sdk/src/codegen/instructions/crankLottoGameClosed.ts
import { TransactionInstruction, PublicKey, AccountMeta } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@coral-xyz/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as types from "../types" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CrankLottoGameClosedArgs {
  round: BN
}

export interface CrankLottoGameClosedAccounts {
  authority: PublicKey
  lottoGame: PublicKey
  eventEmitter: PublicKey
}

export const layout = borsh.struct([borsh.u64("round")])

export function crankLottoGameClosed(
  args: CrankLottoGameClosedArgs,
  accounts: CrankLottoGameClosedAccounts,
  programId: PublicKey = PROGRAM_ID
) {
  const keys: Array<AccountMeta> = [
    { pubkey: accounts.authority, isSigner: true, isWritable: false },
    { pubkey: accounts.lottoGame, isSigner: false, isWritable: true },
    { pubkey: accounts.eventEmitter, isSigner: false, isWritable: true },
  ]
  const identifier = Buffer.from([24, 48, 27, 86, 98, 216, 12, 185])
  const buffer = Buffer.alloc(1000)
  const len = layout.encode(
    {
      round: args.round,
    },
    buffer
  )
  const data = Buffer.concat([identifier, buffer]).slice(0, 8 + len)
  const ix = new TransactionInstruction({ keys, programId, data })
  return ix
}
